// react-native-use-websocket/lib/commonjs/lib/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ReadyState = exports.UNPARSABLE_JSON_OBJECT = exports.DEFAULT_RECONNECT_INTERVAL_MS = exports.DEFAULT_RECONNECT_LIMIT = exports.SOCKET_IO_PING_CODE = exports.SOCKET_IO_PATH = exports.SOCKET_IO_PING_INTERVAL = exports.DEFAULT_OPTIONS = void 0;
const MILLISECONDS = 1;
const SECONDS = 1000 * MILLISECONDS;
const DEFAULT_OPTIONS = {};
exports.DEFAULT_OPTIONS = DEFAULT_OPTIONS;
const SOCKET_IO_PING_INTERVAL = 25 * SECONDS;
exports.SOCKET_IO_PING_INTERVAL = SOCKET_IO_PING_INTERVAL;
const SOCKET_IO_PATH = '/socket.io/?EIO=3&transport=websocket';
exports.SOCKET_IO_PATH = SOCKET_IO_PATH;
const SOCKET_IO_PING_CODE = '2';
exports.SOCKET_IO_PING_CODE = SOCKET_IO_PING_CODE;
const DEFAULT_RECONNECT_LIMIT = 20;
exports.DEFAULT_RECONNECT_LIMIT = DEFAULT_RECONNECT_LIMIT;
const DEFAULT_RECONNECT_INTERVAL_MS = 5000;
exports.DEFAULT_RECONNECT_INTERVAL_MS = DEFAULT_RECONNECT_INTERVAL_MS;
const UNPARSABLE_JSON_OBJECT = {};
exports.UNPARSABLE_JSON_OBJECT = UNPARSABLE_JSON_OBJECT;
let ReadyState;
exports.ReadyState = ReadyState;

(function (ReadyState) {
  ReadyState[ReadyState["UNINSTANTIATED"] = -1] = "UNINSTANTIATED";
  ReadyState[ReadyState["CONNECTING"] = 0] = "CONNECTING";
  ReadyState[ReadyState["OPEN"] = 1] = "OPEN";
  ReadyState[ReadyState["CLOSING"] = 2] = "CLOSING";
  ReadyState[ReadyState["CLOSED"] = 3] = "CLOSED";
})(ReadyState || (exports.ReadyState = ReadyState = {}));
