// react-native-use-websocket/lib/commonjs/lib/use-socket-io.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useSocketIO = void 0;

var _react = require("react");

var _useWebsocket = require("./use-websocket");

var _constants = require("./constants");

const emptyEvent = {
  type: 'empty',
  payload: null
};

const getSocketData = event => {
  if (!event || !event.data) {
    return emptyEvent;
  }

  const match = event.data.match(/\[.*]/);

  if (!match) {
    return emptyEvent;
  }

  const data = JSON.parse(match);

  if (!Array.isArray(data) || !data[1]) {
    return emptyEvent;
  }

  return {
    type: data[0],
    payload: data[1]
  };
};

const useSocketIO = (url, options = _constants.DEFAULT_OPTIONS, connect = true) => {
  const optionsWithSocketIO = (0, _react.useMemo)(() => ({ ...options,
    fromSocketIO: true
  }), []);
  const {
    sendMessage,
    sendJsonMessage,
    lastMessage,
    readyState,
    getWebSocket
  } = (0, _useWebsocket.useWebSocket)(url, optionsWithSocketIO, connect);
  const socketIOLastMessage = (0, _react.useMemo)(() => getSocketData(lastMessage), [lastMessage]);
  return {
    sendMessage,
    sendJsonMessage,
    lastMessage: socketIOLastMessage,
    lastJsonMessage: socketIOLastMessage,
    readyState,
    getWebSocket
  };
};

exports.useSocketIO = useSocketIO;
